app/models/post.rb

```diff
@@ -199,9 +199,9 @@ def read_time_for(viewing_replies)
     return most_recent.updated_at if most_recent.updated_at > edited_at
 
     # testing for case where the post was changed in status more recently than the last reply
-    audits_since_last_reply = audits.where('created_at > ?', most_recent.created_at)
-    audit = audits_since_last_reply.detect { |a| a.audited_changes.key?('status') }
-    return most_recent.updated_at unless audit
+    audits_exist = audits.where('created_at > ?', most_recent.created_at).where(action: 'update')
+    audits_exist = audits_exist.where("(audited_changes -> 'status' ->> 1)::integer = ?", Post::STATUS_COMPLETE).exists?
+    return most_recent.updated_at unless audits_exist
```

Comment:
for viewport reasons I propose:

    audits = audits.where('created_at > ?', most_recent.created_at).where(action: 'update')
    audits = audits_exist.where("(audited_changes -> 'status' ->> 1)::integer = ?", Post::STATUS_COMPLETE).exists?
    return most_recent.updated_at unless audits.exists?

Reply:
...you've got .exists? there twice, but

Reply:
YOU SAW NOTHING NOTHING HAPPENED